src/storage/double_write_buffer.c

```diff
@@ -737,61 +737,53 @@ dwb_adjust_write_buffer_values (unsigned int *p_double_write_buffer_size, unsign
   min_size = DWB_MIN_SIZE;
   max_size = DWB_MAX_SIZE;
 
+  assert (IS_POWER_OF_2 (min_size) && IS_POWER_OF_2 (max_size));
+
   if (*p_double_write_buffer_size < min_size)
     {
       *p_double_write_buffer_size = min_size;
     }
-  else if (*p_double_write_buffer_size > min_size)
+  else if (*p_double_write_buffer_size > max_size)
+    {
+      *p_double_write_buffer_size = max_size;
+    }
+  else if (!IS_POWER_OF_2 (*p_double_write_buffer_size))
     {
-      if (*p_double_write_buffer_size > max_size)
+      unsigned int dwb_limit = min_size << 1;
+
+      while (dwb_limit < DWB_MAX_SIZE && dwb_limit < *p_double_write_buffer_size)
 	{
-	  *p_double_write_buffer_size = max_size;
+	  dwb_limit = dwb_limit << 1;
 	}
-      else
-	{
-	  /* find smallest number multiple of 512 k */
-	  unsigned int limit1 = min_size;
-
-	  while (*p_double_write_buffer_size > limit1)
-	    {
-	      assert (limit1 <= DWB_MAX_SIZE);
-	      if (limit1 == DWB_MAX_SIZE)
-		{
-		  break;
-		}
-	      limit1 = limit1 << 1;
-	    }
 
-	  *p_double_write_buffer_size = limit1;
-	}
+      *p_double_write_buffer_size = dwb_limit;
     }
 
+  assert (IS_POWER_OF_2 (*p_double_write_buffer_size));
+
   min_size = DWB_MIN_BLOCKS;
   max_size = DWB_MAX_BLOCKS;
 
+  assert (IS_POWER_OF_2 (min_size) && IS_POWER_OF_2 (max_size));
```

Comment:
How about replacing `min/max_size` variables with `min/max_block` ?
I think it is more readable, and also is this assertion required? 

if the check is required, I think it is required to be mentioned when defining `DWB_MAX/MIN_BLOCKS` , `DBW_MAX/MIN_SIZE` at the top of `double_writer_buffer.c`.